src/porepy/models/solution_strategy.py

```diff
@@ -228,6 +228,30 @@ def set_discretization_parameters(self) -> None:
 
         """
 
+    def set_solver_statistics(self) -> None:
+        """Set the solver statistics object.
+
+        This method is called at initialization. It is intended to be used to
+        set the solver statistics object(s). Currently, the solver statistics
+        object is related to nonlinearity only. Statistics on linear solvers
+        may be added in the future.
+
+        """
+        # Retrieve the value with a default of pp.SolverStatistics
+        statistics_candidate = self.params.get(
+            "nonlinear_solver_statistics",
+        )
+        # Explicitly check if the retrieved value is a class and a subclass of
+        # pp.SolverStatistics for type checking.
+        if isinstance(statistics_candidate, type) and issubclass(
```

Comment:
Two questions:
1. If I want to use a custom statistics object, this forces me to subclass from `pp.SolverStatistics` and explicitly disallows duck-typing, which I believe is the more pythonic approach. Comments?
2. Is there a reason for not raising an error instead of silently overriding a user's failed attempt at making a custom class? I can see this leading to a lot of confusion.

Reply:
1. I think allowing any (data)class will be difficult, since there is a contract about a save method called by the SolutionStrategy.
2. I think you're right. I'll do as you suggest.